src/Pages.jsx/ViewByTopic.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { getArticleByTopic } from "../Utils/Queries"
+import ArtHeader from "../Components/ArtHeader"
+import ArticleCard from "../Components/ArticleCard"
+
+const ViewByTopic = () =>  { 
+
+    const {topic} = useParams()
+
+    const [articles, setArticles] = useState([])
+    const[isLoading, setIsLoading] = useState(true)
+
+    useEffect(()=> { 
+        getArticleByTopic(topic)
+        .then((res)=> { 
+            setArticles(res)
+            setIsLoading(false)
+        })
+    }, [])
```

Comment:
Toggling between views more than once only works for me if I put topic in the dependency array here! (so that everytime I select a topic from the menu, we refetch the articles and update the view)